recon.py

```diff
@@ -0,0 +1,329 @@
+"""
+A Proposal for reconciliation graph and reconciliation interface.
+"""
+from abc import ABC  # Abstract Base Classes
+from typing import List, Dict
+from enum import Enum
+
+__all__ = ['MappingNode', 'EventType', 'Event', 'Cospeciation', 'Duplication', 'Transfer', 'Loss',
+           'TipTip', 'Reconciliation', 'ReconGraph']
+
+class MappingNode:
+    """
+    MappingNode is a node in the reconciliation graph that maps
+    a parasite to a host.
+    """
+    def __init__(self, parasite_vertex: str, host_vertex: str):
+        self._parasite = parasite_vertex
+        self._host = host_vertex
+
+    @property
+    def parasite(self) -> str:
+        return self._parasite
+
+    @property
+    def host(self) -> str:
+        return self._host
+    
+    def __eq__(self, other):
+        if type(self) == type(other):
+            return self._parasite == other.parasite and self._host == other.host
+        return False
+    
+    def __hash__(self):
+        return hash((self._parasite, self._host))
+    
+    def __repr__(self):
+        return "%s(%s, %s)" % (type(self).__name__, self._parasite, self._host)
+
+class EventType(Enum):
+    """
+    EventType is the type of event node.
+    """
+    COSPECIATION = 1
+    DUPLICATION = 2
+    TRANSFER = 3
+    LOSS = 4
+    TIPTIP = 5
+
+class Event(ABC):
+    """
+    Base Class for events.
+    """
+    def __init__(self, freq: float = None):
```

Comment:
Write a comment that `__init__` cannot be used directly.

Reply:
freq is the only attribute that I can imagine at the moment.  But, it's possible that at some point, we'll discover that we want to associate other attributes with events.  For future extensibility, does it make sense to have a general collection of attributes that we associate with an event?  BTW, we might also want to associate attributes with mapping nodes.

Reply:
I think if we decide to have more attributes in the future, we can just add them to the `__init__` of `Event` and `MappingNode` and then set the corresponding getters `@property` for them.